File: 46. Permutations/permute.java
class backTrack {
    List<List<Integer>> res = new LinkedList<>();

    public void solution(int[] nums, LinkedList<Integer> track) {
        if (nums.length == track.size()) {
            res.add(new LinkedList(track));
            return;
        }
        for (int i: nums) {
            if (track.contains(i)) {
                continue;
            }
            track.add(i);
            backTrack(nums, new LinkedList(track));
            track.removeLast();
        }
    }

    public List<List<Integer>> permute(int[] nums) {
        LinkedList<Integer> track = new LinkedList<>();
        backTrack(nums, track);
        return res;
    }
}

File: 46. Permutations/permute.py
class permute:
    def __init__(self):
        self.res = []

    def backTrack(self, nums, track):
        if len(nums) == len(track):
            self.res.append(track[:])
            return
        for i in nums:
            if i in track:
                continue
            track.append(i)
            self.backTrack(nums, track[:])
            track.pop()
    
    def solution(self, nums: List[int]) -> List[List[int]]:
        track = []
        self.backTrack(nums, track)
        return self.res
    